packages/features/calendar-cache/calendar-cache.repository.ts

```diff
@@ -56,43 +58,144 @@ export class CalendarCacheRepository implements ICalendarCacheRepository {
     return response;
   }
 
-  async getCachedAvailability(credentialId: number, args: FreeBusyArgs) {
-    const key = parseKeyForCache(args);
-    const cached = await prisma.calendarCache.findUnique({
-      where: {
-        credentialId_key: {
-          credentialId,
+  async findUnexpiredUnique({
+    credentialId,
+    delegationCredentialId,
+    userId,
+    key,
+  }: {
+    credentialId: number | null;
+    delegationCredentialId: string | null;
+    userId: number | null;
+    key: string;
+  }) {
+    log.debug("findUnexpiredUnique", safeStringify({ credentialId, delegationCredentialId, userId, key }));
+    const credential = getUserCredential({ userId, delegationCredentialId, credentialId });
+    if (!credential) {
+      return null;
+    }
+    if (credential.type === "delegation") {
+      return prisma.calendarCache.findFirst({
+        where: {
+          delegationCredentialId: credential.delegationCredentialId,
+          userId: credential.userId,
           key,
+          expiresAt: { gte: new Date(Date.now()) },
```

Comment:
We have to think about removing credential check from here so that calendarCache entry is reusable when an organization moves to delegation credential, I will follow up with it.

Reply:
@zomars how about in the followup, we check just `userId`, 'key', `expiresAt` in both delegation-credential and regular credential cases.

This would ensure reusability of cache when switching from Regular Credential to Delegation Credentials(by the way of enabling delegation credentials)

I think `userId` should be enough to authenticate which user's cache it is and as to what calendars cache we are fetching, key would help.